src/hooks/useUSDCPrice.ts

```diff
@@ -78,17 +78,16 @@ export function useStablecoinAmountFromFiatValue(fiatValue: string | null | unde
   const { chainId } = useActiveWeb3React()
   const stablecoin = chainId ? STABLECOIN_AMOUNT_OUT[chainId]?.currency : undefined
 
-  if (fiatValue === null || fiatValue === undefined || !chainId || !stablecoin) {
-    return undefined
-  }
-
   // trim for decimal precision when parsing
-  const parsedForDecimals = parseFloat(fiatValue).toFixed(stablecoin.decimals).toString()
+  const parsedForDecimals =
```

Comment:
Should this be memoized as well?

Reply:
It may be memoized, but it doesn't need to be in order to effect a performance improvement.
I think it's a cheap operation, and the actual re-render is caused by returning distinct objects (from `tryParseCurrencyAmount`), so I'd rather just leave it be.